ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/SourceDescriptionImpl.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.client.impl;
+
+import io.confluent.ksql.api.client.FieldInfo;
+import io.confluent.ksql.api.client.QueryInfo;
+import io.confluent.ksql.api.client.SourceDescription;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+public final class SourceDescriptionImpl implements SourceDescription {
+
+  private final String name;
+  private final String type;
+  private final List<FieldInfo> fields;
+  private final String topic;
+  private final String keyFormat;
+  private final String valueFormat;
+  private final List<QueryInfo> readQueries;
+  private final List<QueryInfo> writeQueries;
+  private final Optional<String> timestampColumn;
+  private final Optional<String> windowType;
+  private final String sqlStatement;
+
+  // CHECKSTYLE_RULES.OFF: ParameterNumberCheck
+  SourceDescriptionImpl(
+      final String name,
+      final String type,
+      final List<FieldInfo> fields,
+      final String topic,
+      final String keyFormat,
+      final String valueFormat,
+      final List<QueryInfo> readQueries,
+      final List<QueryInfo> writeQueries,
+      final Optional<String> timestampColumn,
+      final Optional<String> windowType,
+      final String sqlStatement
+  ) {
+    // CHECKSTYLE_RULES.ON: ParameterNumberCheck
+    this.name = Objects.requireNonNull(name);
+    this.type = Objects.requireNonNull(type);
+    this.fields = Objects.requireNonNull(fields);
+    this.topic = Objects.requireNonNull(topic);
+    this.keyFormat = Objects.requireNonNull(keyFormat);
+    this.valueFormat = Objects.requireNonNull(valueFormat);
+    this.readQueries = Objects.requireNonNull(readQueries);
+    this.writeQueries = Objects.requireNonNull(writeQueries);
+    this.timestampColumn = Objects.requireNonNull(timestampColumn);
+    this.windowType = Objects.requireNonNull(windowType);
+    this.sqlStatement = Objects.requireNonNull(sqlStatement);
```

Comment:
Same comment about using the 2nd parameter to display the variable name on NPE for better troubleshooting. Another reason? say keyFormat is null, it will throw a NPE with line 58. Turns out you saw that exception in ksqlDB 0.12, but master has one new line here, so you'll have to look at 0.12 code to know the right null variable. With the 2nd parameter string will be quickly to find.

Reply:
Done.